File: src/main/java/com/nowsprinting/intellij_mob/service/NotificationService.java
package com.nowsprinting.intellij_mob.service;

import com.intellij.notification.*;
import com.intellij.openapi.components.ServiceManager;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.wm.ToolWindow;
import com.intellij.ui.content.Content;
import org.jetbrains.annotations.NotNull;

public class NotificationService {
    private final NotificationGroup NOTIFICATION_GROUP = new NotificationGroup("Mob", NotificationDisplayType.BALLOON, true);

    public static NotificationService getInstance(@NotNull Project project) {
        return ServiceManager.getService(project, NotificationService.class);
    }

    public Notification information(String content) {
        return notify(null, content, NotificationType.INFORMATION);
    }

    public Notification warning(String content) {
        return notify(null, content, NotificationType.WARNING);
    }

    public Notification error(String content) {
        return notify(null, content, NotificationType.ERROR);
    }

    public Notification notify(Project project, String content, NotificationType type) {
        final Notification notification = NOTIFICATION_GROUP.createNotification(content, type);
        notification.notify(project);
        return notification;
    }

    public void openEventLog(@NotNull Project project) {
        ToolWindow eventLog = EventLog.getEventLog(project);
        if (eventLog != null && !eventLog.isVisible()) {
            eventLog.activate(() -> {
                String contentName = getContentName(NOTIFICATION_GROUP.getDisplayId());
                Content content = eventLog.getContentManager().findContent(contentName);
                if (content != null) {
                    eventLog.getContentManager().setSelectedContent(content);
                }
            }, true);
        }
    }

    @NotNull
    private static String getContentName(@NotNull String groupId) {
        for (EventLogCategory category : EventLogCategory.EP_NAME.getExtensionList()) {
            if (category.acceptsNotification(groupId)) {
                return category.getDisplayName();
            }
        }
        return "";
    }
}

File: src/main/kotlin/com/nowsprinting/intellij_mob/service/NotificationService.kt
package com.nowsprinting.intellij_mob.service

import com.intellij.notification.*
import com.intellij.openapi.components.ServiceManager
import com.intellij.openapi.project.Project

class NotificationService {
    private val NOTIFICATION_GROUP = NotificationGroup("Mob", NotificationDisplayType.BALLOON, true)

    fun information(content: String?): Notification? {
        return notify(null, content, NotificationType.INFORMATION)
    }

    fun warning(content: String?): Notification? {
        return notify(null, content, NotificationType.WARNING)
    }

    fun error(content: String?): Notification? {
        return notify(null, content, NotificationType.ERROR)
    }

    fun notify(
        project: Project?,
        content: String?,
        type: NotificationType?
    ): Notification? {
        val notification = NOTIFICATION_GROUP.createNotification(content!!, type!!)
        notification.notify(project)
        return notification
    }

    fun openEventLog(project: Project) {
        val eventLog = EventLog.getEventLog(project)
        if (eventLog != null && !eventLog.isVisible) {
            eventLog.activate(Runnable {
                val contentName = getContentName(NOTIFICATION_GROUP.displayId)
                val content = eventLog.contentManager.findContent(contentName)
                if (content != null) {
                    eventLog.contentManager.setSelectedContent(content)
                }
            }, true)
        }
    }

    private fun getContentName(groupId: String): String {
        for (category in EventLogCategory.EP_NAME.extensionList) {
            if (category.acceptsNotification(groupId)) {
                return category.displayName
            }
        }
        return ""
    }

    companion object {
        fun getInstance(project: Project): NotificationService? {
            return ServiceManager.getService(
                project,
                NotificationService::class.java
            )
        }
    }
}